server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -0,0 +1,358 @@
+package org.labkey.embedded;
+
+import org.apache.catalina.core.StandardContext;
+import org.apache.catalina.loader.WebappLoader;
+import org.apache.catalina.startup.Tomcat;
+import org.apache.tomcat.util.descriptor.web.ContextResource;
+import org.labkey.bootstrap.ConfigException;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.embedded.tomcat.TomcatWebServer;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.annotation.Validated;
+
+import javax.sql.DataSource;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+@SpringBootApplication
+public class LabKeyServer
+{
+	private static final int BUFFER_SIZE = 4096;
+
+	public static void main(String[] args)
+	{
+		ApplicationContext ctx = SpringApplication.run(LabKeyServer.class, args);
```

Comment:
Indentation looks like it's 8-spaces here instead of our normal 4.

Reply:
Indentation still seems wrong. Is it saved with tabs instead of spaces?

Reply:
interesting.. it doesn't have tabs and looks fine locally.

Reply:
I just pulled down the source and it's using tabs. Try doing View->Active Editor->Show Whitespace when you've got one of these new embedded files open. Also, check the bottom right where it's showing Tab.

![image](https://user-images.githubusercontent.com/5114642/102925506-589dc200-4448-11eb-8dec-badf6e1d7eeb.png)

Reply:
thanks, corrected.